File: core/src/com/colorflow/music/CaptureInterface.java
package com.colorflow.music;

public interface CaptureInterface {

    Type getType();

    byte[] getSamples();

    int getSamplingRate();

    int getCaptureSize();

    int getCaptureRate();

    double getAccurateCaptureRate();

    enum Type {
        PCM, FFT
    }
}

File: core/src/com/colorflow/music/CaptureInterface.kt
package com.colorflow.music

interface CaptureInterface {

    val type: Type

    val samples: ByteArray

    val samplingRate: Int

    val captureSize: Int

    val captureRate: Int

    val accurateCaptureRate: Double

    enum class Type {
        PCM, FFT
    }
}
